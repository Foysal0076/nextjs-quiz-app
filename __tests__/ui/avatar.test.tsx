import Avatar from '@/components/ui/avatar'
import { render, screen } from '@testing-library/react'

describe('Avatar', () => {
  it('renders initials when no URL is provided', () => {
    render(<Avatar name='John Doe' />)
    expect(screen.getByText('JD')).toBeInTheDocument()
  })

  it('renders image when URL is provided', () => {
    const url = 'https://example.com/avatar.jpg'
    render(<Avatar name='John Doe' url={url} />)
    const image = screen.getByAltText('John Doe')
    expect(image).toBeInTheDocument()
    expect(image).toHaveAttribute('src')
  })

  it('applies custom className correctly', () => {
    render(<Avatar name='John Doe' className='custom-class' />)
    expect(screen.getByText('JD').parentElement).toHaveClass('custom-class')
  })

  it('applies custom textClassName to initials', () => {
    render(<Avatar name='John Doe' textClassName='custom-text-class' />)
    expect(screen.getByText('JD')).toHaveClass('custom-text-class')
  })

  // Snapshot test
  it('matches snapshot', () => {
    const { container } = render(<Avatar name='John Doe' />)
    expect(container).toMatchSnapshot()
  })
})
