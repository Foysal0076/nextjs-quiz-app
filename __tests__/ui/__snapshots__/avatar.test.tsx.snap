// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<div>
  <div
    class="relative flex h-8 w-8 items-center justify-center rounded-full border-neutral-100 bg-surface-300 text-base font-bold"
  >
    <span
      class="text-sm font-bold md:text-base"
    >
      JD
    </span>
  </div>
</div>
`;
